### PR TITLE
Remove SetPropertiesDefaults.

### PR DESCRIPTION
**Pull request by yizhang4321**
The only usage of SetPropertiesDefaults in AgentBaker is in cmd/generate.go to support "baker generate" call. We've decided to not have any smarts in setting default values in AgentBaker. The logic about defaults used to be in aks-engine code, now for us the logic resides in RP code base. Having AgentBaker maintain a separate defaults settings that may get out of sync with RP is not good for long term code maintenance. Instead, when using AgentBaker, the caller needs to provide the full ContainerService struct, not depending on AgentBaker setting any defaults.
More code can be removed now that this set defaults func is removed. I'll remove those in future PRs.

### CODE REVIEW

File: cmd/generate.go [hunk 1]
```diff
@@ -14,7 +14,6 @@ import (
 	"github.com/Azure/agentbaker/pkg/aks-engine/api"
```
Comment (AbelHu) [inline cmd/generate.go:14]: I think that we can remove all unused files by AKS to save the effort.

Comment (yizhang4321) [reply cmd/generate.go:14]: Yes, that's the goal for code cleanup. If you see some unused code, feel free to remove it or let me know and I'll remove it.

File: pkg/agent/datamodel/defaults.go [hunk 2]
```diff
@@ -29,36 +29,6 @@ type PropertiesDefaultsParams struct {
 	PkiKeySize int
 }
 
-// SetPropertiesDefaults for the container Properties
-func (cs *ContainerService) SetPropertiesDefaults(params PropertiesDefaultsParams, cloudSpecConfig *AzureEnvironmentSpecConfig) error {
```
Comment (xuto2) [inline pkg/agent/datamodel/defaults.go:33]: I think we can remove the whole defaults.go?

Comment (yizhang4321) [reply pkg/agent/datamodel/defaults.go:33]: Yes, we can. I have just sent this new PR to remove the defaults. https://github.com/Azure/AgentBaker/pull/315

**Pull request merged**